src/phoenix/auth.py

```diff
@@ -33,16 +61,165 @@ def validate_password_format(password: str) -> None:
+        if not password.isascii():
+            raise ValueError("Password must contain only ASCII characters")
+        err_msg = []
+        if len(password) < self.length:
+            err_msg.append(f"must be at least {self.length} characters long")
+        if self.special_chars and not any(char in "!@#$%^&*()_+" for char in password):
+            err_msg.append("at least one special character")
+        if self.digits and not any(char.isdigit() for char in password):
+            err_msg.append("at least one digit")
+        if self.upper_case and not any(char.isupper() for char in password):
+            err_msg.append("at least one uppercase letter")
+        if self.lower_case and not any(char.islower() for char in password):
+            err_msg.append("at least one lowercase letter")
+        if not err_msg:
+            return
+        try:
+            from faker import Faker  # type: ignore[import-not-found,unused-ignore]
+
+            suggested_password = Faker().unique.password(**asdict(self))
+        except BaseException:
+            suggested_password = ""
+        if len(err_msg) > 1:
+            err_text = "Password " + ", ".join(err_msg[:-1]) + ", and " + err_msg[-1]
+        else:
+            err_text = f"Password {err_msg[0]}"
+        if suggested_password:
+            err_text = f"{err_text} (e.g. `{suggested_password}`)"
+        raise ValueError(err_text)
+
+
+"""The name of the header for token-based authentication"""
 EMAIL_PATTERN = re.compile(r"^[^@\s]+@[^@\s]+[.][^@\s]+\Z")
+"""The regular expression pattern for a valid email address."""
 NUM_ITERATIONS = 10_000
+"""The number of iterations to use for the PBKDF2 key derivation function."""
 MIN_PASSWORD_LENGTH = 4
+"""The minimum length of a password."""
+PASSWORD_REQUIREMENTS = _PasswordRequirements(MIN_PASSWORD_LENGTH)
+"""The requirements for a valid password."""
+REQUIREMENTS_FOR_PHOENIX_SECRET = _PasswordRequirements(32, True, True, True, True)
+"""The requirements for the Phoenix secret key."""
+JWT_ALGORITHM = "HS256"
+"""The algorithm to use for the JSON Web Token."""
+PHOENIX_ACCESS_TOKEN_COOKIE_NAME = "phoenix-access-token"
+PHOENIX_ACCESS_TOKEN_COOKIE_MAX_AGE = timedelta(days=31)
+
+
+class Issuer(Enum):
+    API_KEY = auto()
+    SESSION = auto()
+
+
+class ClaimStatus(Enum):
+    VALID = auto()
+    INACTIVE = auto()
+    EXPIRED = auto()
+
+
+@dataclass(frozen=True)
+class _BaseAttributes:
+    user_role: str
+
+
+@dataclass(frozen=True)
+class ApiKeyAttributes(_BaseAttributes):
+    name: Optional[str] = None
+    description: Optional[str] = None
+
+
+@dataclass(frozen=True)
+class SessionAttributes(_BaseAttributes): ...
+
+
+Attributes: TypeAlias = Union[ApiKeyAttributes, SessionAttributes]
+TokenId: TypeAlias = str
+
+
+@dataclass(frozen=True)
+class Claim:
+    token_id: Optional[TokenId] = None
+    user_id: Optional[int] = None
+    issuer: Optional[Issuer] = None
+    issued_at: Optional[datetime] = None
```

Comment:
can you link the RFC 7519 claims for the ones that are meant to be compliant? https://www.rfc-editor.org/rfc/rfc7519#section-4.1.3

Reply:
ok. will do

Reply:
done